README.md

```diff
@@ -1,72 +1,103 @@
 # Yeelight
 
-This project provides rust bindings for yeelight WiFi light interoperation spec.
-[Yeelight_Inter-Operation_spec.pdf][1].
+This project provides Rust bindings for the [Yeelight WiFi Light Interoperation Specification][1].
 
 All the methods specified in the spec are implemented and named equally to the
 aforementioned specification.
 
-This intended as a API bindings for developers, if you want a cli tool to
-control the lights take a look at [yeelight-cli][2] which uses this bindings (It
+This project is intended as API bindings for developers, if you want a CLI tool to
+control the lights take a look at [yeelight-cli][2] which uses these bindings (It
 is also a good example of the usage and capabilities of this crate).
 
-# Usage
+## Table of Contents
+- [Usage](#usage)
+  - [Bulb](#bulb)
+  - [Discovery](#discovery)
+  - [Connection](#connection)
+    - [From discovered bulbs](#from-discovered-bulbs)
+    - [From address](#from-address)
+  - [Basic Operations](#basic-operations)
+  - [Music](#music)
+  - [Flows](#flows)
+- [Implementation](#implementation-details)
+  - [Async](#async)
+  - [Background light](#background-light)
+  - [Features](#features)
+- [Examples](#examples)
+  - [Collatz](#collatz)
+  - [Flow](#flow)
+  - [Music](#music)
+  - [Notification](#notification)
+  - [Properties](#properties)
+  - [Toggle](#toggle)
+- [Roadmap](#roadmap)
+  - [API features](#api-features)
+  - [Quality of life](#quality-of-life)
+  - [Testing](#testing)
+- [Troubleshooting](#troubleshooting)
+  - [Connection refused](#connection-refused)
+
+## Usage
 
 The usage is quite straight forward, you can use the built-in bulb discovery
 method to locate the Bulbs and connect to them. You can also connect directly
 if you know the address beforehand. Once connected to the Bulb you can call the
 various methods to change it's status.
 
-## Bulb
+### Bulb
 
-The [Bulb] object represents an active connection to a singular light. All
+The `Bulb` object represents an active connection to a singular light. All
```

Comment:
[Bulb] is so that `cargo doc` links it with the proper location in the crate documentation. I agree though that it looks quite ugly in GitHub Markdown.

Maybe the best option is to use [`Bulb`] which works in both cargo and GitHub README.